### DISCUSSION TITLE
WebApi Solution with PhoneNumber instead of email

### DISCUSSION THREAD

**Discussion by KothariDhrumil** [JonPSmith/AuthPermissions.AspNetCore#117] [Ideas]
1. Hi @JonPSmith 

First off, huge congratulations on creating something truly extraordinary! I'm also sorry to hear about your illness.

John, I’d love to expand your library to help me develop my own product (SaaS-based).

Product definition:

1. The product will primarily focus on inventory management. I plan to implement a multi-tenant system (though a single tenant will suffice for now) with sharding for each individual firm.
2. The firm owner will act as the Tenant Admin.
3. This firm owner can add their drivers, who will follow a specific route, serve customers, deliver products (like vegetables), and handle cash transactions. This can be managed using filter queries such as tenant ID and route ID.
4. I’m considering different versions: a free version for a single route and driver, a Small Scale Version for three routes and three drivers, and so on.
5. All functionalities will be accessible through mobile apps (a customer app for placing orders and checking transactions), a Driver App (for delivering orders on specific routes and reporting), and an admin website.

So far, I’m not too worried as I believe it can be managed using AuthP.

Here are some challenges I foresee:

1. Customers need to be able to interact across tenants. For instance, Customer A should be able to purchase items from Tenant A, B, and C using just one application for all three. I think you’ve addressed this issue before, and I’ll need to implement it. (https://github.com/JonPSmith/AuthPermissions.AspNetCore/issues/91). In the mobile app, customers will have the option to select a tenant once they are added by the Tenant Admin.
   
2. I need to adjust the logic to check if a user already exists and simply attach the tenant if they do.
   
3. All users should register using their mobile numbers instead of email addresses.

I would appreciate your guidance on this. Please let me know how and where I can get started.

**Comment by JonPSmith** [comment]
Hi @KothariDhrumil,

Let me answer your challenge items first:

1. **Customers interact across tenants**. You could that by using a hierarchical multi-tenant application, and the top level would be for the customers, with the firms at the second level. You would need to have a frontend for the customers which shows a list of firms to buy from.
2.  **Check user already exists**: AuthP does that automatically.
3. **Mobile numbers instead of email addresses**. ASP.NET Core needs emails - I don't think you can change that (I think). But you could also ask for their mobile number on login and have code to use it. 

Taking about your drivers that can work for multiple firms the drivers would log in the top level hierarchical level, like the customers do but the drivers are sent to the "Driver" section, like the one in your number 5 item. Because the drivers are at the top level they can access any firms. Of course you would add code that which firms works for and only shows those firms. 

I hope that helps.
